src/controllers/userController.ts

```diff
@@ -0,0 +1,60 @@
+import { Request, Response } from 'express';
+import User from '../models/userModel';
+
+export const getAllUsers = async (req: Request, res: Response): Promise<void> => {
+  const { page = 1, limit = 10 } = req.query;
+
+  try {
+    const users = await User.find()
+      .limit(Number(limit) * 1)
+      .skip((Number(page) - 1) * Number(limit))
+      .exec();
+    
+    const count = await User.countDocuments().exec();
+    
+    res.status(200).json({
```

Comment:
when building rest-api always return only actual data, no meta data. You should do res.json(users) and that would be sufficient. If you want to sent how many pages there are (or even better - how many items in a collection) use headers pair:
Frontend sends: Return-Total-Count: 'true'
Backend returns: X-Total-Count: 'actual-count'

this is the cleanest communication